src/Database/Table/SqlBuilder.php

```diff
@@ -552,6 +552,16 @@ protected function addWhereComposition(array $columns, array $parameters)
 		}
 	}
 
+	private function getConditionHash($condition, $parameters) {
+		foreach ($parameters as &$parameter) {
+			if ($parameter instanceof Selection) {
+				$parameter = $this->getConditionHash($parameter->getSql(), $parameter->sqlBuilder->getParameters());
+			} elseif (is_object($parameter) && method_exists($parameter, '__toString')) {
```

Comment:
The same problem as with Selection can be with another object (for example with ActiveRow or some entity class build on ActiveRow). I have add another test for active row...